src/ray/common/ray_syncer/ray_syncer-inl.h

```diff
@@ -0,0 +1,201 @@
+// Copyright 2022 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace ray {
+namespace syncer {
+
+/// NodeState keeps track of the modules in the local nodes.
+/// It contains the local components for receiving and reporting.
+/// It also keeps the raw messages receivers got.
+class NodeState {
+ public:
+  /// Constructor of NodeState.
+  NodeState();
+
+  /// Set the local components.
+  ///
+  /// \param cid The component id.
+  /// \param reporter The reporter is defined to be the local module which wants to
+  /// broadcast its internal status to the whole clsuter. When it's null, it means there
+  /// is no reporter in this node for this component. This is the place there messages are
+  /// generated.
+  /// \param receiver The receiver is defined to be the module which eventually
+  /// will have the view of of the cluster for this component. It's the place where
+  /// received messages are consumed.
+  ///
+  /// \return true if set successfully.
+  bool SetComponents(RayComponentId cid,
```

Comment:
nit: SetComponent? since we are setting one component here. Although I'm not sure how it's being used, but seems it's more like AddComponent or RegisterComponent?